maven-scm-client/src/main/java/org/apache/maven/scm/client/cli/MavenScmCli.java

```diff
@@ -121,36 +121,6 @@ public static void main( String[] args )
         scmUrl = args[2];
 
         ScmVersion scmVersion = null;
-        if ( args.length > 3 )
-        {
-            String version = args[3];
-
-            if ( args.length > 4 )
-            {
-                String type = args[4];
-
-                if ( "tag".equals( type ) )
-                {
-                    scmVersion = new ScmTag( version );
-                }
-                else if ( "branch".equals( type ) )
-                {
-                    scmVersion = new ScmBranch( version );
-                }
-                else if ( "revision".equals( type ) )
-                {
-                    scmVersion = new ScmRevision( version );
-                }
-                else
-                {
-                    throw new IllegalArgumentException( "'" + type + "' version type isn't known." );
-                }
-            }
-            else
-            {
-                scmVersion = new ScmRevision( args[3] );
-            }
-        }
```

Comment:
Why is this whole block gone?

Reply:
HI @michael-o 

Because `args.length` can never > 3  otherwise will return error in line 105

` if ( args.length != 3 )`

Reply:
You have found a bug. it should have been `<` and not `!=`. Last two args are optional.

Reply:
HI @michael-o 
I still don't know what the best solution is. For the moment i made the rollback of this part of the code.
TY

Reply:
Report the bug for now and go on with this PR.